### PR TITLE
refactor: replaced package/bem with @textkernel/bem package

### PR DESCRIPTION
**Pull request by antipin**
**Breaking changes / new features in bem library:**

image

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ Try to make components as descriptive as possible to increase accessibility for
...
+OneUI comes with a [BEM-utility](https://github.com/textkernel/bem) that auto-generates and applies CSS classes on the basis of the component name and its props that affect styling. Detailed instructions with regard to how to use the utility [can be found here](https://github.com/textkernel/bem/blob/master/README.md). The most important rule is that any CSS property values that should be affected by themes need to be defined as CSS variables. Each CSS variable should come with a default value, defined by the base theme. Any static CSS property values _will not be affected_ when using custom themes.
```
[the remark this answers is not in the document]
Comment (antipin) [reply CONTRIBUTING.md]: good point

File: scripts/test/jest.config.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ module.exports = {
 
     // An array of directory names to be searched recursively up from the requiring module's
     // location
-    moduleDirectories: ['node_modules', 'src/packages'],
+    moduleDirectories: ['node_modules', 'src', 'src/packages'],
```
[the remark this answers is not in the document]
Comment (antipin) [reply scripts/test/jest.config.js]: I'd rather keep src only (see this thread https://github.com/textkernel/oneui/pull/185#discussion_r342607796)

File: src/components/Alert/Alert.js [hunk 3]
```diff
@@ -1,15 +1,11 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import bem from 'bem';
+import bem from 'utils/bem';
```
[the remark this answers is not in the document]
Comment (antipin) [reply src/components/Alert/Alert.js]: > Do you have such shortcuts in Search?
we do

I also can live with both, but for myself I figured out kind of a rule:
Either keep project structure flat and use relative imports, or (if you tend to embrace nesting), aliases and absolute imports saves a lots of time, especially if you decided to move components around on a filesystem.

File: package.json [hunk 4]
```diff
@@ -47,6 +47,7 @@
     "dependencies": {
         "@babel/polyfill": "7.0.0",
         "@react-google-maps/api": "1.5.4",
+        "@textkernel/bem": "0.2.4",
```
[the remark this answers is not in the document]
Comment (eszthoff) [reply package.json]: Probably not. It is something that npm discovers every once in awhile. However you get the warning when you run `npm i` and it takes only 1 extra minute to resolve these issues. So I think it makes sense to run it whenever you see it / when you make any changes to `package.lock`. making separate PRs just for that is more headache

[the remark this answers is not in the document]
Comment (antipin) [reply package.json]: Mmm.. I don't think it's a good practice. Updating deps that do not relate to a story:
1. Can confuse reviewer
2. Can lead to broken builds/tests.

In search we do it as a separate PRs usually

File: src/components/Alert/Alert.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (antipin) [reply src/components/Alert/Alert.js]: @igorarkhipenko I like the idea with `~` to make it clear that we import not from modules but from the root of the project

[the remark this answers is not in the document]
Comment (antipin) [reply src/components/Alert/Alert.js]: But for now I'll keep relative imports cause it's not part of the story indeed and later (e.g. in TS story) we can discuss it and come up with a conclusion.

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (antipin) [reply package.json]: Indeed, this time it complains only about deeper dependencies. Fixed :)

File: src/components/Alert/Alert.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (eszthoff) [reply src/components/Alert/Alert.js]: I also like the `~` approach, let's discuss that in the TS story.

**Pull request merged**